internal/server/boltdbstate/job.go

```diff
@@ -749,6 +752,13 @@ func (s *State) JobAck(id string, ack bool) (*serverstate.Job, error) {
 	}
 
 	txn.Commit()
+
+	// Update the task state machine if acked job is part of an on-demand runner task.
+	if err := s.TaskAck(job.Id); err != nil {
```

Comment:
Does this (and TaskComplete) need to be an exported API? Or can this just be a detail of the job system and unexported?

Reply:
Agreed - Yeah this can live inside the job API details as unexported functions since they're an implementation detail of how the job API works! Moving now and updating my tests accordingly.

Reply:
Covered in 3859b4a0f9fda70bbe8834735cde26d29979eca4 with some extra job tests for the task job triple lifecycle.